src/elements/filter-menu.html

```diff
@@ -0,0 +1,235 @@
+            [[resultsCount]] {$ filters.results $}
+          </div>
+        </div>
+
+        <div class="actions" layout horizontal center>
+        <span class="reset-filters" role="button" on-tap="_resetFilters" hidden$="[[!_selectedArray.length]]">
+          {$ filters.clear $}
+        </span>
+          <paper-button class="icon-right" on-tap="_toggleBoard">
+            {$ filters.title $}
+            <iron-icon icon="hoverboard:[[_getFilterIcon(_openedBoard)]]"></iron-icon>
+          </paper-button>
+        </div>
+      </div>
+
+      <div class="selected-filters" hidden$="[[!_selectedArray.length]]">
+        <template is="dom-repeat" items="[[_selectedArray]]" as="selectedFilter">
+          <div
+            class="tag"
+            style$="--color: [[getVariableColor(selectedFilter.value, 'primary-text-color')]]"
+            filter-key$="[[selectedFilter.key]]"
+            filter-value$="[[selectedFilter.value]]"
+            on-tap="_toggleFilter"
+            selected
+            layout horizontal inline center
+          >
+            <span>[[selectedFilter.value]]</span>
+            <iron-icon icon="hoverboard:close"></iron-icon>
+          </div>
+        </template>
+      </div>
+    </div>
+
+    <div class="filters-board" block$="[[_openedBoard]]">
+      <div class="container">
+        <template is="dom-repeat" items="[[filters]]" as="filter">
+          <div class="filter-group">
+            <h3 class="filter-title">[[filter.title]]</h3>
+            <template is="dom-repeat" items="[[filter.items]]" as="item">
+              <div
+                class="tag"
+                style$="--color: [[getVariableColor(item, 'primary-text-color')]]"
+                filter-key$="[[filter.key]]"
+                filter-value$="[[item]]"
+                selected$="[[_isSelected(selected, filter.key, item)]]"
+                on-tap="_toggleFilter"
+              >
+                [[item]]
+              </div>
+            </template>
+          </div>
+        </template>
+      </div>
+    </div>
+
+
+  </template>
+
+  <script>
+
+    // eslint-disable-next-line no-undef
+    class FilterMenu extends UtilsFunctions(Polymer.Element) {
+      static get is() {
+        return 'filter-menu';
+      }
+
+      static get properties() {
+        return {
+          filters: Array,
+          resultsCount: Number,
+          selected: {
+            type: Object,
+            value: {},
+          },
+          _selectedArray: {
+            type: Array,
+            computed: '_generateSelectedArray(selected)',
+          },
+          _openedBoard: {
+            type: Boolean,
+            value: false,
+          },
+        };
+      }
+
+      _isSelected(selectedFilters, key, value) {
+        return selectedFilters[key]
+          && selectedFilters[key].includes(this.generateClassName(value.trim()));
+      }
+
+      _toggleFilter(e) {
+        const filterKey = e.currentTarget.getAttribute('filter-key');
```

Comment:
Would be good to save 'e.currentTarget' to a variable, as you've used it a few times. Also, check if it does exist, otherwise 'return' from method

Reply:
only 2 times, doesn't make sense to add one more line for that :)

Reply:
but make sense to check if a target is not null

Reply:
You click on the element, the target will always exist